internal/pkg/manifest/transform.go

```diff
@@ -18,58 +21,284 @@ var basicKinds = []reflect.Kind{
 	reflect.Complex64, reflect.Complex128,
 	reflect.Array, reflect.String, reflect.Slice,
 }
+var fmtExclusiveFieldsSpecifiedTogether = "invalid manifest: %s %s mutually exclusive with %s and shouldn't be specified at the same time"
 
 type workloadTransformer struct{}
 
 // Transformer returns custom merge logic for workload's fields.
 func (t workloadTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
-	if typ == reflect.TypeOf(Image{}) {
-		return transformImage()
+	if transform := t.transformExclusiveTypes(typ); transform != nil {
+		return transform
 	}
 
-	if typ == reflect.TypeOf(EntryPointOverride{}) {
-		return transformStringSliceOrString(reflect.TypeOf(EntryPointOverride{}))
+	if transform := t.transformCompositeTypes(typ); transform != nil {
+		return transform
 	}
 
-	if typ == reflect.TypeOf(CommandOverride{}) {
-		return transformStringSliceOrString(reflect.TypeOf(CommandOverride{}))
+	if typ.String() == "map[string]manifest.Volume" {
+		return transformMapStringToVolume
 	}
 
-	if typ == reflect.TypeOf(Alias{}) {
-		return transformStringSliceOrString(reflect.TypeOf(Alias{}))
+	if transform := transformBasic(typ); transform != nil {
+		return transform
 	}
 
-	if typ.String() == "map[string]manifest.Volume" {
-		return transformMapStringToVolume()
+	return nil
+}
+
+func (t workloadTransformer) transformCompositeTypes(typ reflect.Type) func(dst, src reflect.Value) error {
+	var et exclusiveTransformer
+	switch typ {
+	case reflect.TypeOf(EntryPointOverride{}), reflect.TypeOf(CommandOverride{}), reflect.TypeOf(Alias{}):
```

Comment:
What do you think of this to loose the coupling 💭
```suggestion
	switch reflect.New(typ).Elem() {
	case val.CanConvert(reflect.TypeOf(stringSliceOrString{})):
```

Reply:
updated!